File: 4_Tópicos_avançados/4.08_OpenCV/Ex10/Media_linhas_colunas.cpp
// Baseado em https://docs.opencv.org/master/db/df5/tutorial_linux_gcc_cmake.html

#include <stdio.h>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;

int main(int argc, char** argv)
{
	if(argc != 2)
	{
		printf("Modo de uso: %s Nome_Imagem\n",
			argv[0]);
		return -1;
	}
	Mat image = imread(argv[1]);
	Mat media_colunas(image.rows,
		image.channels(), CV_32S, Scalar(0));
	Mat media_linhas(image.channels(),
		image.cols, CV_32S, Scalar(0));
	int y, x, c, S;
	for(y=0; y<image.rows; y++)
		for(c=0; c<image.channels(); c++)
		{
			S=0;
			for(x=0; x<image.cols; x++)
				S += image.at<Vec3b>(y, x)[c];
			media_colunas.at<int>(y, c) =
				(S+image.cols/2)/image.cols;
		}
	for(x=0; x<image.cols; x++)
		for(c=0; c<image.channels(); c++)
		{
			S=0;
			for(y=0; y<image.rows; y++)
				S += image.at<Vec3b>(y, x)[c];
			media_linhas.at<int>(c, x) =
				(S+image.rows/2)/image.rows;
		}
	cout << " ==> Imagem: " << argv[1] << endl;
	cout << " Linhas: " << media_linhas << endl;
	cout << " Colunas: " << media_colunas << endl;
	return 0;
}

File: 4_Dicas/4.1_OpenCV/Ex10/Media_linhas_colunas.py
import cv2, numpy, sys

if len(sys.argv)<2:
	print("Modo de uso: " + sys.argv[0] + " Nome_Imagem")
	sys.exit(-1)
image = cv2.imread(sys.argv[1])
rows,cols,channels = image.shape

media_linhas  = numpy.round(numpy.mean(image, 0))
media_colunas = numpy.round(numpy.mean(image, 1))

print(" ==> Imagem: " + sys.argv[1])
print(" Linhas: ")
print(media_linhas)
print(" Colunas: ")
print(media_colunas)
